client-mr/core/src/main/java/org/apache/hadoop/mapreduce/RssMRUtils.java

```diff
@@ -228,27 +231,33 @@ public static String getString(Configuration rssJobConf, String key, String defa
   }
 
   public static long getBlockId(int partitionId, long taskAttemptId, int nextSeqNo) {
```

Comment:
The assertions in this method are redundant, calling `LAYOUT.getBlockId` provides them already. All this method need to do is call

    return LAYOUT.getBlockId(nextSeqNo, partitionId, taskAttemptId);

Same in tez client.